src/messages/message.ts

```diff
@@ -0,0 +1,15 @@
+export abstract class Message {
+    public constructor(
+        protected id: string,
+        protected name: string,
+    ) {
+    }
+
+    protected abstract generate(type: string, parameters: object): string;
+}
+
+const generateParameters = (parameters: object): string =>
+    Object.entries(parameters).map(([key, value]) => `${key}='${value || ''}'`).join(' ')
+
+export const generateMessage = (type: string, flowId: string, parameters: object = {}) =>
+    `##teamcity[${type} flowId='${flowId}' ${generateParameters(parameters)}]`
```

Comment:
Not sure if this would be idiomatic typescript but could this be a protected method of the message class?

Reply:
> Not sure if this would be idiomatic typescript but could this be a protected method of the message class?

No problem